src/metabase/util/malli/fn.clj

```diff
@@ -154,27 +153,16 @@
                                                   [{:keys [value message]}]
                                                   (str message ", got: " (pr-str value)))})
             details   (merge
-                        {:type      error-type
-                         :error     error
-                         :humanized humanized
-                         :schema    schema
-                         :value     value}
-                        error-context)]
-        (if (or config/is-dev?
-              config/is-test?)
```

Comment:
This is redundant because it will always be `true`. `validate` is always downstream of a `mu.fn/instrument-fn?` call,  which already checks that `(or config/is-dev? config/is-test?)` is `true`:
https://github.com/metabase/metabase/blob/8ba09e810b0397943d281a6504e6814c1b1a03ae/src/metabase/util/malli/fn.clj#L290-L295

So the `false` branch of this `if` has been dead code since https://github.com/metabase/metabase/pull/37428

Reply:
It also checks that `(true? (:instrument/always (meta namespace)))`, so if you are in a ns with that metadata and in prod, then `instrument-fn?` will be true, and `(or config/is-dev? config/is-test?)` will be `false`.

Reply:
That seemed like a bug to me, not an intentional choice behind the meaning of `:instrument/always`. We don't use `:instrument/always` so this is all philosophical, but I thought the meaning of that should be "instrument in prod as we do in dev".

Reply:
I think its worth discussing, but without consensus it's kind of hard to change something like that outright. 

IMO it's better to log warnings for errors in prod (the github thing) than crash the program due to a malli check failure. It's quite possible that the validation check fails _but the program would have continued working_, and we had a bug in the malli schema.

Reply:
Fair point, I think I'll close the PR at this stage. No one is using `:instrument/always` yet, and yes, it could be used in the future so this code could be handy. I don't need to spend anyone's time getting consensus on theoretical scenarios where we do use it